src/main/java/frc/robot/Subsystems/ClimberPlacer.java

```diff
@@ -1,50 +1,32 @@
 package frc.robot.Subsystems;
 
 import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
-import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
-import edu.wpi.first.wpilibj.DigitalInput;
-import frc.robot.RobotMap;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+
+import java.util.function.Supplier;
 
 /**
  * Controls the position of the climber.
  */
 public class ClimberPlacer extends MotoredGenericSubsystem {
 
-    private static final double MIN_SPEED = -0.6;
-    private static final double MAX_SPEED = 0.6;
-
-    private static ClimberPlacer instance;
-
-    private final DigitalInput frontLimit;
-    private final DigitalInput backLimit;
-    private static WPI_TalonSRX leftPlacer;
-    private static WPI_TalonSRX rightPlacer;
-
-    public static ClimberPlacer getInstance() {
-        if (instance == null) {
-            leftPlacer = new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_LEFT);
-            rightPlacer = new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_RIGHT);
-            instance = new ClimberPlacer(leftPlacer, rightPlacer);
-        }
-        return instance;
+    public static ClimberPlacer getLeftInstance() {
+        return null;
     }
 
-    private ClimberPlacer(WPI_TalonSRX leftPlacer, WPI_TalonSRX rightPlacer) {
-        super(MIN_SPEED, MAX_SPEED, "climber placer", leftPlacer, rightPlacer);
-        this.frontLimit = new DigitalInput(RobotMap.DIO.PLACER_LIMIT_FRONT);
-        this.backLimit = new DigitalInput(RobotMap.DIO.PLACER_LIMIT_BACK);
+    public static ClimberPlacer getRightInstance() {
+        return null;
     }
 
-    @Override
-    public boolean canMove(double speed) {
-        return !(ClimberWinch.getInstance().isHooked() || speed > 0 && frontLimit.get() || speed < 0 && backLimit.get());
+    public ClimberPlacer(Supplier<Double> minSpeed, Supplier<Double> maxSpeed, String namespaceName, MotorController... motorControllers) {
```

Comment:
constructor should go above getInstance(), which should also go below isStalling() (convention A.III.a.3)